backupccl: ensure makeSimpleImportSpans handles span frontier checkpointing

msbutler opened the pull request:
If bulkio.restore.use_simple_import_spans is true on 23.1, we previously only used the high water mark to checkpoint things. But the high water mark checkpointing is version gated not feature gated. This leads to a real bad bug: on 23.1, if bulkio.restore.use_simple_import_spans is set to true, a restore cannot read the persisted span frontier, implying it would have to redo all work again. This patch ensures that makeSimpleImportSpans can read from the span frontier.

Additional testing for this will come in #99304

Informs #98779

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/99373)
<!-- Reviewable:end -->

msbutler committed:
backupccl: ensure makeSimpleImportSpans handles span frontier checkpointing

If bulkio.restore.use_simple_import_spans is true on 23.1, we previously only
used the high water mark to checkpoint things. But the high water mark
checkpointing is version gated not feature gated. This leads to a real bad bug:
on 23.1, if bulkio.restore.use_simple_import_spans is set to true, a restore
cannot read the persisted span frontier, implying it would have to redo all
work again. This patch ensures that makeSimpleImportSpans can read from the
span frontier.

Additional testing for this will come in #99304

Informs #98779

Release note: None

msbutler commented:
TFTR!

bors r=rhu713
